### ISSUE TITLE
sw carpernty

### ISSUE THREAD

**Issue by floswald** [HugoLhuillier/IntroProg2018#1]
* https://swcarpentry.github.io/shell-novice/
* https://github.com/bmatzelle/gow/wiki for a shell
* git

## which language?

* R just works
* julia is a real language
* efficiency is not first order concern
* maybe both?
* maybe translate this to julia? https://swcarpentry.github.io/r-novice-inflammation/
* big issue with julia: user friendliness. which IDE to use?

**Comment by floswald** [comment]
a different way to think about this is:

* R is hard to beat for people who will mostly be doing applied work. little simulation here and there. great graphics, data, statistics, and nonparametric.
* julia is more for people who will actually have to solve complicated models. (an altogether different route would be to teach Rcpp for those people , i.e. have a pure R course that ends up with calling C++ functions; but we prefer julia to overcome this two-language (R+Cpp) problem).
* it's related to the *reduced form vs structural model* division
* I don't think we should foster this division, as it misses the point. But if you are honest, there will be 70% reduced form guys and 30% others in your course. 
* the question is how to maximize aggregate welfare of this group of students. maybe 70% R, 30% julia? just thinking out loud here...

**Comment by HugoLhuillier** [comment]
I see your point. However, they might be confuse if they have to learn two languages, you might have some welfare loss that applies to both groups here. But I don't know if it dominates...

If we go for the 2-languages approach, what do you think about this kind of schedule?

1. What's a computer? How to interact with it? Introduction to Unix and Github 
2. Computing: the basics in R (data type, variable, logical operators, functions etc.). Basically, most of what's in https://swcarpentry.github.io/r-novice-inflammation/ and what's needed if you want to estimate your own estimator or stuff like that
3. Advanced computing in Julia (creating your own type, package management, code profiling and optimization etc.)

So if everything goes well, it would be something like 1-2 lectures on 1), 3-4 lectures on 2), and 2-3 lectures on 3). And then adjusting depending on how responsive the students are.

**Comment by HugoLhuillier** [comment]
(also if we only teach them Julia, maybe none of them will go for reduced form economics. and that's a big :tada:. kidding!)

**Comment by floswald** [comment]
yes sounds good to me. i would focus a lot on the swcarpentry part. keep in mind that we care mostly about the students with 0 programming knowledge, not whether the top programmer (or indeed the teacher!) is bored. :-) like that first section is really fundamental. assigning to a variable etc. i would use ggplot right after base plot, it's just too powerful to ignore.

**Comment by HugoLhuillier** [comment]
I'm a bit rusty in R myself anyway, so that's going to be a good training! Thanks for your expertise Florian!

**Comment by floswald** [comment]
definitely good training. R is a bit awkward sometimes, you know, just like a 20-year old language that grew out of C and S-plus would look like. but there are amazing things there, particularly for data work.  look at what hadley wickham did (apart from ggplot): https://www.tidyverse.org. also, talking about efficiency: someone still will have to beat R data.table: https://github.com/Rdatatable/data.table/wiki/Benchmarks-%3A-Grouping. and - contrarily to common wisdom amongst economists - stata is **not** more efficient with large datasets than R: https://github.com/matthieugomez/benchmark-stata-r

I just started a project with spatial data. impossible to beat R. anyway, i'm happy to look at all your material and provide comments. also, copy at will from tyler's stuff if you want.
